### ISSUE TITLE
Serializing on big vs. little endian systems

### ISSUE THREAD

**Issue by mschubert** [armstrtw/rzmq#19]
There is an option in R's `serialize()` function that determines whether data is serialized in big- or little endian format. Big endian is default, which doesn't make much sense, because most current systems use little endian. This has a performance penalty that could easily be avoided by setting the option `xdr=FALSE` in [rzmq.R#L52](https://github.com/armstrtw/rzmq/blob/52a491defab74d235fb04ded707502eb0c6dbdee/R/rzmq.R#L52) (but could be queried using `.Platform$endian` or combined to `xdr=.Platform$endian=="big"`).

```r
ll = lapply(1:1e6, function(x) as.list(setNames(runif(5), letters[1:5])))

fx = function(ll, xdr) { 
    serial = serialize(ll, NULL, xdr)
    unserial = unserialize(serial)
}                      

system.time(fx(ll, xdr=TRUE))
system.time(fx(ll, xdr=FALSE))
```

Which gives the following times on my institute's computing facilities:

```r
> system.time(fx(ll, xdr=TRUE))
   user  system elapsed
 25.124   1.246  26.633
> system.time(fx(ll, xdr=FALSE))
   user  system elapsed
  2.873   0.315   3.279
```

**Comment by armstrtw** [comment]
Interesting. Yes, big endian is an artifact of sun and other risc based architectures early dominance.

I'm fine with providing an option for it possibly sourced from getOption. Or if you have it set up, send a PR.

**Comment by mschubert** [comment]
merged in #20
